core/src/bundle/usBundleRegistry.cpp

```diff
@@ -59,10 +59,17 @@ void BundleRegistry::Clear()
   bundles.v.clear();
 }
 
-std::vector<Bundle> BundleRegistry::Install(const std::string& location, BundlePrivate* caller)
+std::vector<Bundle> BundleRegistry::Install(const std::string& location,
+                                            BundlePrivate* caller,
+                                            bool allowExecutable)
 {
   CheckIllegalState();
-
+  
+  if (!allowExecutable && !IsSharedLibrary(location))
+  {
+    throw std::runtime_error("Installing bundle at "+ location + " not permitted");
```

Comment:
It would be nice to communicate the real reason why the install failed, e.g.
> Cannot install bundles from \<location\> because it is not a shared library